make make_fx collective test single threaded

ydwu4 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #133731
* __->__ #134775

make_fx is not thread-safe due to mutating and patching global states. It's difficult and low roi to make it thread-safe so just turn the tracing test into a single-thread test.

cc @XilunWu @H-Huang @awgu @kwen2501 @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @c-p-i-o

ydwu4 committed:
make make_fx collective test single threaded

[ghstack-poisoned]

yifuwang reviewed (approved):
It's not the most ideal to set world_size=1 for distributed tests. It would be legit to fast-path collectives above the dispatcher layer when world_size=1 in which case the test would break (though we are not doing it here).

I would suggest switching to fake_pg for compile + distribtued tests that don't need to check results (such as this one). It's very simple to use ([example](https://github.com/pytorch/pytorch/blob/main/test/distributed/test_c10d_functional_native.py#L490-L496))

ydwu4 committed:
Update on "make make_fx collective test single threaded"


make_fx is not thread-safe due to mutating and patching global states. It's difficult and low roi to make it thread-safe so just turn the tracing test into a single-thread test.

cc XilunWu H-Huang awgu kwen2501 wanchaol fegin fduwjj wz337 wconstab d4l3k c-p-i-o

[ghstack-poisoned]

ydwu4 commented:
Updated the pr with fake_pg!
